src/tokio.rs

```diff
@@ -379,15 +379,24 @@ where
     client_async_tls_with_connector_and_config(request, socket, connector, config).await
 }
 
-use pin_project::pin_project;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
-/// Adapter for `tokio::io::AsyncRead` and `tokio::io::AsyncWrite` to provide
-/// the variants from the `futures` crate and the other way around.
-#[pin_project]
-#[derive(Debug, Clone)]
-pub struct TokioAdapter<T>(#[pin] pub T);
+pin_project_lite::pin_project! {
+    /// Adapter for `tokio::io::AsyncRead` and `tokio::io::AsyncWrite` to provide
+    /// the variants from the `futures` crate and the other way around.
+    #[derive(Debug, Clone)]
+    pub struct TokioAdapter<T> {
+        #[pin]
+        inner: T,
+    }
+}
+
+impl<T> TokioAdapter<T> {
+    pub(crate) fn new(inner: T) -> Self {
```

Comment:
```suggestion
    pub fn new(inner: T) -> Self {
```

It's needed for providing custom tokio `AsyncRead`/`AsyncWrite` as the underlying stream of the websocket.